src/main/resources/org/jenkinsci/plugins/github/config/GitHubPluginConfig/config.groovy

```diff
@@ -27,10 +27,11 @@ f.section(title: descriptor.displayName) {
             f.entry(title: _("Override Hook URL")) {
                 g.blockWrapper {
                     f.optionalBlock(title: _("Specify another hook URL for GitHub configuration"),
+                            name: "isOverrideHookUrl",
                             inline: true,
                             checked: instance.isOverrideHookUrl()) {
                         f.entry(field: "hookUrl") {
-                            f.textbox(checkMethod: "post")
+                            f.textbox(checkMethod: "post", name: "hookUrl")
                         }
                     }
                 }
```

Comment:
For example all of this should just be like

```xml
<f:entry field="hookUrl">
  <f:textbox label="Alternate hook URL for GitHub configuration"/>
</f:entry>
```

which may be blank.

Reply:
Oh I see. Remove all the optional block that is not necessary.. Yeah sounds good to me I can test this.

Reply:
Would be more work for sure. The usual problem with this sort of fix is retaining compatibility for both XML settings and JCasC. Sometimes hacking up the JSON processing in the GUI logic is the only option since refactoring to a simpler structure would break something.

Reply:
the thing that bother me is the type `URL` of the `hookUrl`... Because of that type, the value cannot actually be blank :( :

```
java.net.MalformedURLException: no protocol: 
	at java.base/java.net.URL.<init>(URL.java:645)
	at java.base/java.net.URL.<init>(URL.java:541)
	at java.base/java.net.URL.<init>(URL.java:488)
	at org.kohsuke.stapler.Stapler$3.convert(Stapler.java:1141)
Caused: org.apache.commons.beanutils.ConversionException: no protocol: 
```

So it is not that simple to "reset" the hook URL because of the binding.. I thought about changing the type from `URL` to `String` but the existing public getter might be a problem: https://github.com/jenkinsci/github-plugin/blob/master/src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java#L131-L141

Changing its signature could be a breaking change. Though it does not seem to be used outside the scope of that plugin repo: https://github.com/search?ref=simplesearch&type=code&q=user%3Ajenkinsci++getHookUrl.

Reply:
BTW use permalinks for better display: https://github.com/jenkinsci/github-plugin/blob/47b623aad58baffe8c2d20c0ada1339124229804/src/main/java/org/jenkinsci/plugins/github/config/GitHubPluginConfig.java#L131-L141